### DISCUSSION TITLE
23.02

### DISCUSSION THREAD

**Discussion by falkTX** [DISTRHO/Cardinal#501] [Announcements]
Cardinal 23.02

<hr /><em>This discussion was created from the release <a href='https://github.com/DISTRHO/Cardinal/releases/tag/23.02'>23.02</a>.</em>

**Comment by subderground** [comment]
Hi,
I think i find a bug (that close Reaper  PC 64bit).
If I make and save a Cardinal preset with the last 22.12 version that includes Surge XT modules inside. It isn't possible to open it anymore with the 23.02 version (VST3 or LV2). 
I have not tested all Surges XT modules (But to check and identify the problematic module in my preset, I've finally tried a new preset with only S&H Noize and Filter = crash/bug).

**Reply by falkTX** [reply]
What OS are you using? Windows or Linux?

@baconpaul does this ring a bell? rack patches with surge-xt 2.0, loaded after updating to surge-xt 2.1
I do not get a crash on macOS Intel at least.

**Reply by baconpaul** [reply]
Nope in rack we are patch compatible across the 2.0 to 2.1 without problem

**Reply by falkTX** [reply]
ok thanks. will need to investigate. I think it only happens on Windows...

**Reply by baconpaul** [reply]
the majority of the streaming changes 2.0 -> 2.1 are consistently using json_set_new rather than json_set to make sure we don't leak references. I wonder if cardinal load path has a reference difference from rack? maybe that's where to look.

we've had zero crashing stream reports in vcvrack though

**Reply by falkTX** [reply]
since it does not happen on macOS (tested myself) or Linux (tested by @dromer), I think it is some very odd case special to Windows..
will investigate later

**Reply by subderground** [reply]
No problem, I only have 2 presets made for now with Surge Xt Modules. I can recreate it.

**Comment by benaldhouse** [comment]
This is great. Thanks so much for this. I got a firewall message when I installed it. Is there some new communications feature in this version?

**Reply by falkTX** [reply]
kinda, there is something brewing behind the scenes to allow to control a cardinal instance remotely.
it is not fully fleshed out yet, but already tested to work in some scenarios. will finish that up properly for 23.04 probably

**Comment by GBaige** [comment]
Any reason why the CPU usage meter is missing from the top right in this version (23.02)?

**Comment by 88KeysForHire** [comment]
Today I upgraded three Ubuntu Studio 22.10 machines to 23.04.  Every Surge XT module I try to load crashes Cardinal.  The same modules in VCV Rack have no problems, and all other brand modules I've tried in Cardinal have been fine.  Carla 2.5.4, Cardinal 23.02, Ubuntu Studio 23.04.  Tried with Jack, ALSA, and Pulse:  .so, .lv2, and .vst versions all crash Cardinal trying to load Surge XT modules.  Happy to help troubleshoot if I can.  I can reproduce this on all three upgraded machines.

**Reply by 88KeysForHire** [reply]
Confirmed that it is all the Surge VCO modules that crash Cardinal.  I loaded every other Surge module with no trouble.  Pick a random Surge VCO and it crashes the whole Cardinal session.

**Reply by GBaige** [reply]
This was reported in issue #502, and it seems to have to do with whether the binaries were built with LTO enabled or otherwise. For instance Surge XT modules present no issues in build a42ca2c; in 2352682 LTO was re-enabled to test this theory, and all VCO modules from Surge XT crash once more.

What is LTO? No idea.

**Reply by dromer** [reply]
LTO = Link Time Optimization. You can read more about it here -> https://gcc.gnu.org/wiki/LinkTimeOptimization

**Comment by houyun08** [comment]
hey, Ive been trying to download on my Mac but it keeps sayin the dev is unidentified.
is there something I'm doing wrong? any info welcomed

**Reply by falkTX** [reply]
this is normal, the builds are purposefully unsigned.
you need to right click the pkg and select open
